### PR TITLE
feat: Sort repo page

### LINKED ISSUE #8582: [FEATURE] Sort for repos page
### Description

Currently repos are sorted by most recently active, this is good for default, but additional options would be useful

- sort by stars
- sort by forks
- sort by push date
- sort by created date

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sital002**
## Fixes Issue

closes #8582 

## Changes proposed
Sort feature:
- sort by stars
- sort by forks
- sort by push date
- sort by created date

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
image
image

### CODE REVIEW

File: components/form/Select.js [hunk 1]
```diff
@@ -8,9 +8,12 @@ export default function Select({
   ...restProps
 }) {
   return (
-    <>
+    <div>
       {label && (
-        <label htmlFor={name} className={`${className} hidden md:block`}>
+        <label
+          htmlFor={name}
+          className={`dark:text-white ${className} hidden md:block`}
```
Comment (eddiejaoude) [inline components/form/Select.js:15]: Can you use `className` function https://github.com/EddieHubCommunity/BioDrop/blob/main/services/utils/classNames.js

File: pages/repos.js [hunk 2]
```diff
@@ -23,7 +46,21 @@ export default function Repos({ repos }) {
...
         <h1 className="text-4xl mb-4 font-bold">Community Repos</h1>
-
+        <div className=" flex justify-end">
+          <Select
+            name="event-type"
+            value={router.query.sortBy || "pushed-date"}
+            label="Sort by"
+            onChange={(e) => {
+              router.push(`/repos?sortBy=${e.currentTarget.value}`);
+            }}
```
Comment (eddiejaoude) [inline pages/repos.js:56]: I think this doesn't need the `{}` wrappers?

```suggestion
            onChange={(e) =>  router.push(`/repos?sortBy=${e.currentTarget.value}`)}
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍 

I left some small inline comments

File: pages/repos.js (hunk 2 above)
Comment (eddiejaoude) [reply pages/repos.js]: This was marked as resolved but I don't see any changes or comment? 🤔  I will mark as unresolved

File: pages/repos.js [hunk 3]
```diff
@@ -23,7 +46,21 @@ export default function Repos({ repos }) {
 
       <Page>
         <h1 className="text-4xl mb-4 font-bold">Community Repos</h1>
-
+        <div className=" flex justify-end">
+          <Select
+            name="event-type"
+            value={router.query.sortBy || "pushed-date"}
+            label="Sort by"
+            onChange={(e) =>
```
Comment (sital002) [inline pages/repos.js:54]: I have removed the ```{}```

File: pages/repos.js [hunk 4]
```diff
@@ -23,7 +46,21 @@ export default function Repos({ repos }) {
...
+              router.push(`/repos?sortBy=${e.currentTarget.value}`)
+            }
```
Comment (sital002) [inline pages/repos.js:56]: I have removed the ```{}``` did that on same commit.

Comment (eddiejaoude) [reply pages/repos.js:56]: Sorry my mistake 🤦‍♂️

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Pull request merged**